coordinator/src/test/scala/filodb.coordinator/queryplanner/MultiPartitionPlannerSpec.scala

```diff
@@ -923,94 +923,82 @@ class MultiPartitionPlannerSpec extends AnyFunSpec with Matchers with PlanValida
-    queryParams1.startSecs shouldEqual startSeconds
-    queryParams1.endSecs shouldEqual endSeconds
-    queryParams1.stepSecs shouldEqual step
-    remoteExec1.queryContext.plannerParams.processFailure shouldEqual true
-    remoteExec1.queryContext.plannerParams.processMultiPartition shouldEqual false
-    remoteExec1.queryEndpoint shouldEqual "remote-url1"
-
-    val remoteExec2 = stitchRvsExec.children(1).asInstanceOf[PromQlRemoteExec]
-    val queryParams2 = remoteExec1.queryContext.origQueryParams.asInstanceOf[PromQlQueryParams]
-    queryParams2.startSecs shouldEqual startSeconds
-    queryParams2.endSecs shouldEqual endSeconds
-    queryParams2.stepSecs shouldEqual step
-    remoteExec2.queryContext.plannerParams.processFailure shouldEqual true
-    remoteExec2.queryContext.plannerParams.processMultiPartition shouldEqual false
-    remoteExec2.queryEndpoint shouldEqual "remote-url2"
-
-  }
-
-  it ("should generate second Exec with start and end time equal to query end time when query duration is less" +
-    "than or equal to lookback ") {
-
-    val startSeconds = 1594309980L
-    val endSeconds = 1594310280L
-    val localPartitionStartMs: Long = 1594309980001L
-    val step = 15L
-
-    val partitionLocationProvider = new PartitionLocationProvider {
-      override def getPartitions(routingKey: Map[String, String], timeRange: TimeRange): List[PartitionAssignment] = {
-        if (routingKey.equals(Map("job" -> "app"))) List(
-          PartitionAssignment("remote1", "remote-url", TimeRange(startSeconds * 1000 - lookbackMs,
-            localPartitionStartMs - 1)), PartitionAssignment("remote2", "remote-url",
-            TimeRange(localPartitionStartMs, endSeconds * 1000)))
-        else Nil
-      }
-
-      override def getMetadataPartitions(nonMetricShardKeyFilters: Seq[ColumnFilter], timeRange: TimeRange): List[PartitionAssignment] = List(
-        PartitionAssignment("remote1", "remote-url", TimeRange(startSeconds * 1000 - lookbackMs,
-          localPartitionStartMs - 1)), PartitionAssignment("remote2", "remote-url",
-          TimeRange(localPartitionStartMs, endSeconds * 1000)))
-
-    }
-    val engine = new MultiPartitionPlanner(partitionLocationProvider, localPlanner, "local", dataset, queryConfig)
-    val lp = Parser.queryRangeToLogicalPlan("test{job = \"app\"}", TimeStepParams(startSeconds, step, endSeconds))
-
-    val promQlQueryParams = PromQlQueryParams("test{job = \"app\"}", startSeconds, step, endSeconds)
-
-    val execPlan = engine.materialize(lp, QueryContext(origQueryParams = promQlQueryParams, plannerParams =
-      PlannerParams(processMultiPartition = true)))
-    val stitchRvsExec = execPlan.asInstanceOf[StitchRvsExec]
-    stitchRvsExec.children.size shouldEqual (2)
-    stitchRvsExec.children(0).isInstanceOf[PromQlRemoteExec] shouldEqual (true)
-    stitchRvsExec.children(1).isInstanceOf[PromQlRemoteExec] shouldEqual (true)
-
+    execPlan.isInstanceOf[PromQlRemoteExec] shouldEqual (true)
```

Comment:
Why should it generate just one? The previous one was generating two children,  both for different partition endpoints.

Reply:
Nope, they were generated for the same endpoints:
```
    val remoteExec = stitchRvsExec.children(0).asInstanceOf[PromQlRemoteExec]
    ...
    remoteExec.queryEndpoint shouldEqual "remote-url"

    val remoteExec2 = stitchRvsExec.children(1).asInstanceOf[PromQlRemoteExec]
    ...
    remoteExec2.queryEndpoint shouldEqual "remote-url"
```
This test used to generate two children because one child plan was materialized per shard-key.

Reply:
Some additional comments: this test needed an update because -- now that all split plans (plans with at least one shard-key that resides in >1 partitions) are materialized via `materializePlanHandleSplitLeaf`-- an empty result is returned when all timestamp evaluations require data from >1 partitions.

That would happen here:
- This is an instant query, so only one timestamp is evaluated at `t=1000s`.
- The lookback is `100s`.
- The partition split happens at `950s`.

I've pushed the start of the second partition back far enough to fit the entire `100s` lookback, and now we should only expect one `RemoteExec` to be returned.

Reply:
Got it, so the method would return ``EmptyResultExec`` if left unchanged? If yes why not keep that test and additionally test your case too which goes to just one partition for a timestamp >= 1050?

Reply:
Good point -- I will re-add the `EmptyResultExec` test.